src/main/java/org/jenkins_ci/plugins/run_condition/core/CauseCondition.java

```diff
@@ -66,15 +68,17 @@ public boolean isCausedBy(String className) {
 
         // if XTrigger plugin is installed:
         // file change
-        FS_CAUSE("org.jenkinsci.plugins.fstrigger.FSTriggerCause", "FSTrigger"),
+        FS_CAUSE(XTriggerCause.class, "FSTrigger"),
```

Comment:
Hi,

In fact, @baritonehands you are right.
If you look at the build xml, you get this:
```
<?xml version='1.0' encoding='UTF-8'?>
<build>
  <actions>
    <org.jenkinsci.lib.xtrigger.XTriggerCauseAction plugin="fstrigger@0.39">
      <build class="build" reference="../../.."/>
    </org.jenkinsci.lib.xtrigger.XTriggerCauseAction>
    <hudson.model.CauseAction>
      <causes>
        <org.jenkinsci.lib.xtrigger.XTriggerCause plugin="fstrigger@0.39">
          <triggerName>FSTrigger</triggerName>
          <causeFrom>Triggered by a change to a file</causeFrom>
          <logEnabled>true</logEnabled>
        </org.jenkinsci.lib.xtrigger.XTriggerCause>
      </causes>
    </hudson.model.CauseAction>
  </actions>
</build>

```
So the plugin is actually not working with the plugins linked to XTrigger.

Also note that the **isCausedBy** method has been updated to take the trigger name into account, which is the second part of the correction for xtrigger causes.
As this creates the needs the xtrigger **lib** (not the xtrigger plugin), this makes no sense to stay with a string here, IMHO. Of course, reflection is still an option.

Regards,

Arnaud